File: javascript/jsfind/src/filetype.js
/*
 * filetype.js
 *
 * FileType "static" class
 *
 */

const FileType = {
    UNKNOWN: 0,
    ARCHIVE: 1,
    AUDIO:   2,
    BINARY:  3,
    CODE:    4,
    FONT:    5,
    IMAGE:   6,
    TEXT:    7,
    VIDEO:   8,
    XML:     9
};
Object.freeze(FileType);

exports.FileType = FileType;

File: java/javafind/src/main/java/javafind/FileType.java
package javafind;

public enum FileType {
    UNKNOWN("unknown"),
    ARCHIVE("archive"),
    AUDIO("audio"),
    BINARY("binary"),
    CODE("code"),
    FONT("font"),
    IMAGE("image"),
    TEXT("text"),
    VIDEO("video"),
    XML("xml");

    private final String name;

    FileType(final String name) {
        this.name = name;
    }

    public static FileType forName(final String name) {
        final var lName = name.trim().toLowerCase();
        for (final var fileType : FileType.values()) {
            if (fileType.name.equals(lName)) {
                return fileType;
            }
        }
        return FileType.UNKNOWN;
    }

    public String toName() {
        return this.name;
    }
}
